src/sentry/uptime/subscriptions/subscriptions.py

```diff
@@ -268,14 +293,26 @@ def update_project_uptime_subscription(
             owner_team_id = owner.id
             owner_user_id = None
 
+    env = default_if_not_set(uptime_monitor.environment, environment)
     uptime_monitor.update(
-        environment=default_if_not_set(uptime_monitor.environment, environment),
+        environment=env,
         name=default_if_not_set(uptime_monitor.name, name),
         mode=mode,
         owner_user_id=owner_user_id,
         owner_team_id=owner_team_id,
     )
 
+    if detector := get_detector(uptime_monitor.uptime_subscription):
```

Comment:
Ahh tricky, at first I didn't realise this was inherently filtering out the none values

Reply:
that's almost the entire point of this syntax right